### PR TITLE
node-fetch: make implementeation compatible with lib.dom.d.ts

### PR DESCRIPTION
**Pull request by kammerjaeger**
This should make the implementation compatible to: https://github.com/microsoft/TypeScript/blob/3fc5f968ca002562508bb1ffb863b8b5e2df4492/lib/lib.dom.d.ts#L1926

This should fix the problem from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44649

Maybe there is a better way to test this change.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/microsoft/TypeScript/blob/3fc5f968ca002562508bb1ffb863b8b5e2df4492/lib/lib.dom.d.ts#L1926

### CODE REVIEW

**Review by AndrewLeedham — APPROVED** [review]

File: types/node-fetch/node-fetch-tests.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ function test_fetchUrlWithRequestObject() {
             }) => undefined,
 
             dispatchEvent: (event: any) => false,
-            onabort: null,
+            onabort: (event: string) => false,
```
Comment (OmgImAlexis) [inline types/node-fetch/node-fetch-tests.ts:70]: This should be an additional test as null is still allowed.

File: types/node-fetch/node-fetch-tests.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ function test_fetchUrlObjectWithRequestObject() {
             }) => undefined,
 
             dispatchEvent: (event: any) => false,
-            onabort: null,
+            onabort: (event: any) => "something",
```
[the remark this answers is not in the document]
Comment (kammerjaeger) [reply types/node-fetch/node-fetch-tests.ts]: yes you are right, I changed it

**Review by OmgImAlexis — APPROVED** [review]

**Review by nikcorg — APPROVED** [review]

**Review by AndrewLeedham — APPROVED** [review]

**Review by kbkk — APPROVED** [review]
Well done.

I've encountered this countless times but cba fixing it 👍 💯

**Review by jakebailey — APPROVED** [review]

**Comment by OmgImAlexis** [comment]
Ready to merge

**Comment by OmgImAlexis** [comment]
Ready to merge

**Pull request merged**